src/main/java/org/primefaces/extensions/component/letteravatar/LetterAvatarRenderer.java

```diff
@@ -71,18 +95,56 @@ public void encodeMarkup(FacesContext context, LetterAvatar letterAvatar) throws
             writer.writeAttribute("style", style, "style");
         }
 
-        writer.writeAttribute("width", width, null);
-        writer.writeAttribute("height", height, null);
-        writer.writeAttribute("avatar", avatar, null);
+        writer.writeAttribute("title", value, null);
+
+        writer.startElement("span", letterAvatar);
+        writer.writeAttribute("class", "ui-letteravatar-initials", null);
+        writer.writeAttribute("style", styleSpan(size), null);
+        writer.write(initials);
+        writer.endElement("span");
+
+        writer.endElement("div");
+    }
 
-        writer.endElement("img");
+    private int hash(String str) {
+        int result = 0;
+        for (int i = 0; i < str.length(); i++) {
+            result = ((result << 5) - result) + str.charAt(i);
+            result |= 0;
+        }
+        return result;
+    }
+
+    private int hue(String str) {
+        return Math.abs(hash(str) % 360);
```

Comment:
Don't we need some modulo here to prevent colors that are too close to other colors?

Reply:
I tried to convert your algorithm from javascript to java :) is it wrong?

Reply:
I'm not saying it is wrong. I'm suggesting an improvement.